specifyweb/frontend/js_src/lib/components/TreeView/Actions.tsx

```diff
@@ -117,7 +119,7 @@ export function TreeViewActions<SCHEMA extends AnyTree>({
           disabled={focusedRow === undefined}
           isRoot={isRoot}
           label={
-            hasTablePermission(tableName, 'update')
+            isViewMode
               ? commonText.edit()
               : commonText.view()
```

Comment:
Currently, the button says Edit instead of View when read only mode is turned on